app/screens/home/search/results/file_result.tsx

```diff
@@ -0,0 +1,159 @@
+
+export type XyOffset = {x: number; y: number} | undefined;
+
+const styles = StyleSheet.create({
+    container: {
+        flex: 1,
+        marginHorizontal: 20,
+    },
+});
+
+type Props = {
+    action: GalleryAction;
+    canDownloadFiles: boolean;
+    channelName: string | undefined;
+    fileInfo: FileInfo;
+    index: number;
+    isSingleImage: boolean;
+    lastViewedFileInfo: FileInfo | undefined;
+    onPress: (idx: number) => void;
+    publicLinkEnabled: boolean;
+    setAction: (action: GalleryAction) => void;
+    setLastViewedFileInfo: (fInfo: FileInfo) => void;
+    updateFileForGallery: (idx: number, file: FileInfo) => void;
+}
+
+const galleryIdentifier = 'search-files-location';
+
+const FileResult = ({
+    action,
+    canDownloadFiles,
+    channelName,
+    fileInfo,
+    index,
+    isSingleImage,
+    lastViewedFileInfo,
+    onPress,
+    publicLinkEnabled,
+    setAction,
+    setLastViewedFileInfo,
+    updateFileForGallery,
+}: Props) => {
+    const elementsRef = useRef<View | null>(null);
+    const theme = useTheme();
+    const isTablet = useIsTablet();
+    const isReplyPost = false;
+    const insets = useSafeAreaInsets();
+    const numOptions = useNumberItems(canDownloadFiles, publicLinkEnabled);
+
+    const [showOptions, setShowOptions] = useState<boolean>(false);
+    const [openUp, setOpenUp] = useState<boolean>(false);
+    const [xyOffset, setXYoffset] = useState<XyOffset>(undefined);
+    const {height} = Dimensions.get('window');
+
+    const fileRef = useCallback((element: View) => {
+        if (showOptions) {
+            elementsRef.current = element;
+            elementsRef?.current?.measureInWindow((x, y) => {
+                setOpenUp((y > height / 2));
+                setXYoffset({x, y});
+            });
+        }
+    }, [elementsRef, showOptions]);
+
+    const handleSetAction = useCallback((a: GalleryAction) => {
+        setAction(a);
+        if (showOptions && a !== 'none') {
+            setShowOptions(false);
+        }
+    }, [action, setAction, showOptions]);
+
+    const handleOpenOptions = useCallback((fInfo: FileInfo) => {
+        setShowOptions(true);
+        setLastViewedFileInfo(fInfo);
+
+        if (!isTablet) {
+            showMobileOptionsBottomSheet({
+                fileInfo,
+                insets,
+                numOptions,
+                setAction: handleSetAction,
+                theme,
+            });
+        }
+    }, [isTablet, numOptions, showOptions, theme]);
+
+    // This effect handles the case where a user has the FileOptions Modal
+    // open and the server changes the ability to download files or copy public
+    // links. Reopen the Bottom Sheet again so the new options are added or
+    // removed.
+    useEffect(() => {
+        if (lastViewedFileInfo === undefined) {
+            return;
+        }
+        if (NavigationStore.getNavigationTopComponentId() === Screens.BOTTOM_SHEET) {
```

Comment:
Check corner case for mobile. Filter is open or team is open and admin changes publiclink or download changes. what happens.